exonum-java-binding/core/src/main/java/com/exonum/binding/runtime/ServiceLoader.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.runtime;
+
+import java.net.URI;
+import java.util.Optional;
+
+/**
+ * A dynamic loader of Exonum service artifacts. It allows to load and unload service artifacts and
+ * keeps track of already loaded services.
+ */
+interface ServiceLoader {
+
+  /**
+   * Loads the service artifact. The loading also involves verification of the artifact, e.g.,
+   * checking that it has correct metadata. The actual verifications are implementation-specific.
+   *
+   * <p>Upon successful completion of this method, the service is considered as loaded
+   * and instances of this service can be created.
+   *
+   * @param artifactLocation the location of the service artifact (a file, network resource, etc.)
+   * @return a complete definition of a service from the artifact
+   * @throws ServiceLoadingException in case the artifact cannot be loaded: not found or
+   *     does not pass verification
+   */
+  LoadedServiceDefinition loadService(URI artifactLocation) throws ServiceLoadingException;
```

Comment:
Looks like it'll be convert problems at `Path artifactPath = Paths.get(artifactLocation);`. 
So,
1) By using URI you need to specify local paths in `file:///foo/bar` format
2) it'll fail if you pass some another correct URI like `http://foo.bar`

Reply:
> you need to specify local paths in file://

Not really, `/foo/bar` works just fine.

> it'll fail if you pass some another correct URI like

Yep, it will. I wonder if we shall add some (downloading to temp folder :thinking:) code to handle that. I don't think we shall complicate things till there is a need to do that. Shall we instead
1. Specify that loaders may throw if they do not support an URI scheme?
2. Require a local path to a file?